### DISCUSSION TITLE
Setting the Android Push Notification Icon background color

### DISCUSSION THREAD

**Discussion by ManuelRauber** [firebase/flutterfire#13327] [Q&A]
At the moment, when my Push notification arrives, it looks like this:

![image](https://github.com/user-attachments/assets/e5a0769c-807c-45ca-a814-63c4cc3bb6d2)

As you can see, the red background color does not fill the available background.

My foreground icon is a transparent image containing the white "m". 
The background color comes from

```xml
<resources>
    <color name="ic_launcher_background">#FF0000</color>
</resources>
```

I wonder why this color is picked up at all.

In my `AndroidManifest.xml`, I additionally added this, located in the `<application>` tag but not in a `<activity>` tag:

```xml
<meta-data
            android:name="com.google.firebase.messaging.default_notification_color"
            android:value="#00ABCD" />
```

But it is not replacing the color. 

My goal is that the red background color fills the background, like other notifications do it with their icons as well.

The notification itself does not contain any color or icon information, so the fallback should be whatever is defined in the manifest.

**Comment by imahsanali** [comment]
facing this same issue but in my case its blue as my app primary color is blue, also this issue is only coming upward android 11

**Comment by atila-b** [comment]
Same issue here. Already set AndroidManifest.xml and color parameter in AndroidNotificationDetails.

**Comment by atila-b** [comment]
I found a solution. Create the notification icon on the Android Asset Studio website, unzip and copy and paste the folders to the `android/app/src/main/res` directory. I also needed to add this code to `AndroidManifest.xml`: 
```
<meta-data
            android:name="com.google.firebase.messaging.default_notification_icon"
            android:resource="@drawable/ic_stat_icon_foreground" />
```

**Comment by davidsamaila610-design** [comment]
<resources>
    <color name="ic_launcher_background">#FF0000</color>
</resources>
